pkg/server/worker.go

```diff
@@ -29,41 +41,41 @@ func (s *Server) worker(name string, h *host.Host) {
 			s.logger.Infof("Worker for host %s: Latency=%v (Ping successful)", name, latency)
 			h.Alive = true
 			h.Latency = latency
-			// Update the RRD file with the fetched latency and timestamp
-			s.logger.Debugf("Worker for host %s: Updating RRD with latency %f microseconds.", name, float64(latency.Microseconds()))
-			err = rrdFile.SafeUpdate(time.Now(), []float64{float64(latency.Microseconds())})
-			if err != nil {
-				s.logger.Errorf("Worker for host %s: Failed to update RRD (%v)", name, err)
-			} else {
-				s.logger.Debugf("Worker for host %s: RRD update successful.", name)
-			}
+			latencyUpdate = []float64{float64(latency.Microseconds())}
+		}
+		// Update the RRD file with the fetched latency and timestamp
+		// If the slice is empty, SafeUpdate will handle it
+		s.logger.Debugf("Worker for host %s: Updating RRD with latency %f microseconds.", name, float64(latency.Microseconds()))
+		err = rrdFile.SafeUpdate(time.Now(), latencyUpdate)
+		if err != nil {
+			s.logger.Errorf("Worker for host %s: Failed to update RRD (%v)", name, err)
+		} else {
+			s.logger.Debugf("Worker for host %s: RRD update successful.", name)
 		}
 	}
 
-	// Add a random delay of 1-59 seconds before starting
-	startDelay := time.Duration(rand.Intn(59)+1) * time.Second
-	s.logger.Debugf("Worker for host %s will start in %v", name, startDelay)
-	select {
-	case <-time.After(startDelay):
-		// Perform the initial ping
-		performPing()
-	case <-s.done:
-		s.logger.Infof("Worker for host %s received shutdown signal before starting", name)
-		return
-	}
+	// Perform the initial ping before the ticket starts
+	performPing()
 
 	// Run periodic pings every minute
-	ticker := time.NewTicker(1 * time.Minute)
-	defer ticker.Stop()
-
 	for {
 		select {
-		case <-ticker.C:
-			// Perform the periodic ping
-			performPing()
 		case <-s.done:
 			s.logger.Infof("Worker for host %s received shutdown signal.", name)
 			return
+		default:
+			// Perform the work synchronously
+			performPing()
+
+			// Once the work is done, wait for the interval.
+			// If a shutdown signal arrives during this wait, we exit early.
+			select {
+			case <-time.After(time.Minute):
```

Comment:
I switched from a `ticker` to `sleep` style (and simplified the mutexes) because the cloud instance I was testing on was falling behind. Technically not necessary but will make this a polite neighbor when running along side all the critical services next week.